cocos/2d/CCNode.h

```diff
@@ -1881,9 +1906,16 @@ class CC_DLL Node : public Ref
     mutable bool _additionalTransformDirty; ///< transform dirty ?
     bool _transformUpdated;         ///< Whether or not the Transform object was updated since the last frame
 
-    int _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+#if CC_64BITS
+    std::int64_t _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+#else
+    int _localZOrder; /// < Local order (relative to its siblings) used to sort the node
+    unsigned int _orderOfArrival; /// < used to preserve sequence while sorting children with the same localZOrder
+#endif
```

Comment:
which means _localZOrderAndArrival is a cache value, but used to optimize for 64 bit.